src/Configuration/src/ConfigServerBase/ConfigServerConfigurationProvider.cs

```diff
@@ -264,6 +264,8 @@ internal ConfigEnvironment DoLoad(bool updateDictionary = true)
                             "Located environment: {name}, {profiles}, {label}, {version}, {state}", env.Name, env.Profiles, env.Label, env.Version, env.State);
                         if (updateDictionary)
                         {
+                            Data.Clear();
```

Comment:
@dtillman I won't add this one here, this could impact requests which are currently reading the dictinoary value. I would add another local temp dictonary and add the new values init and replace with existing one .

Reply:
Ahh .. great point .. thanks!
Dave